landlab/components/uniform_precip/generate_uniform_precip.py

```diff
@@ -361,10 +432,109 @@ def yield_storm_interstorm_duration_intensity(self,
                     yield (interstorm_duration, 0.)
                 self._elapsed_time += interstorm_duration
 
+    def yield_storms(self):
+        """Iterator for a time series of storm-interstorm pairs.
+
+        This method is very similar to this component's other generator,
+        yield_storm_interstorm_duration_intensity(), but the way it yields is
+        slightly different. Instead of yielding (interval_duration, rf_rate),
+        with interstorms represented as intervals with rf_rate = 0, it yields:
+
+            (storm_duration, interstorm_duration)
+
+        When each tuple pair is yielded, the grid scalar field 'rainfall__flux'
+        is updated with the rainfall rate occuring during storm_duration.
+
+        This generator method is designed for direct equivalence with the
+        spatially resolved generators found elsewhere in Landlab.
+
+        This generator will be useful in cases where the whole sequence of
+        storms and interstorms doesn't need to be stored, where we can save
+        memory this way.
+
+        This method does not attempt to subdivide timesteps. If you want that,
+        provide a delta_t at instantiation, and use
+        yield_storm_interstorm_duration_intensity(subdivide_interstorms=True).
+
+        The method will keep yielding until it reaches the RUN_TIME, where it
+        will terminate. If it terminates during a storm, the final tuple will
+        be (truncated_storm_duration, 0.). Otherwise it will be
+        (storm_duration, truncated_interstorm_duration.)
+
+        Yields
+        ------
+        tuple of float
+            (storm_duration, interstorm_duration)
+
+        Notes
+        -----
+        One recommended procedure is to instantiate the generator, then call
+        instance.next() (in Python 2) or next(instance) (in Python 3)
+        repeatedly to get the sequence.
+
```

Comment:
I wrote `??` since I don't know what to put... though later on, I saw some nice examples in the code...